### PR TITLE
Add third-party app directory

### PR DESCRIPTION
**Pull request by LeKeve**
## Description

This PR sets up an app directory widget for third party applications. The widget will live in the Learning Center and request app data from Okta through an app directory endpoint. 

**This widget is a WIP. It is not currently connected to the app directory api endpoint.**

Ticket: https://vajira.max.gov/secure/RapidBoard.jspa?rapidView=777&view=detail&selectedIssue=API-2212

Original Design Guidelines: https://github.com/department-of-veterans-affairs/va.gov-team/issues/9238

New issue/design guidelines: https://github.com/department-of-veterans-affairs/va.gov-team/issues/14279

## Testing done

- Local testing of widget on static pages
- Unit tests

The widget can be tested locally by mounting to a static page that has an existing widget id. For example, to mount this widget to the `/find-forms` page
```
// src/applications/static-pages/widgetTypes.js
...
FIND_VA_FORMS: 'find-va-forms', // change to 'anything'
...
THIRD_PARTY_APP_DIRECTORY: 'third-party-app-directory', // change to 'find-va-forms'
...
```

## Screenshots
Screen Shot 2020-09-09 at 6 05 57 AM

## Acceptance criteria
- [x] Initial POC for app directory widget

## Definition of done
- [] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by reedship — COMMENTED** [review]
Looks good minus some DRY stuff.

File: src/applications/third-party-app-directory/components/SearchResult/index.jsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        </div>
+
+        {/* Toggle More Info */}
+        <div className="learn-more">
+          <button
+            className="va-button-link vads-u-text-decoration--none vads-u-border-color--link-default vads-u-border-style--dotted vads-u-border-bottom--1px vads-u-margin-top--1p5"
+            onClick={() => setShow(!show)}
+            type="button"
+          >
+            Learn about {item?.name} <i className="fa fa-cenvron-down" />
```
[the remark this answers is not in the document]
Comment (bradenshipley) [reply src/applications/third-party-app-directory/components/SearchResult/index.jsx:86]: dont forget to fix typo here at 'fa-cenvron-down' >> 'fa-chevron-down' and add functionality to show 'fa-chevron-up' when drop down is enabled

**Comment by rianfowler** [comment]
I suggest updating the [CODEOWNERS](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/codeowners.md) config in a separate PR and merging it in this branch before pushing this out of draft.

**Review by rianfowler — COMMENTED** [review]
What's the reason for loading this through the static pages app?

**Comment by LeKeve** [comment]
> What's the reason for loading this through the static pages app?

@rianfowler The widget will be mounted to a static page in the Learning Center. The original design docs don't reflect this and new designs are being prepared in the following issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/14279

**Review by rianfowler — APPROVED** [review]

**Review by reedship — APPROVED** [review]

**Pull request merged**